app/policies/application_policy.rb

```diff
@@ -40,7 +40,7 @@ def destroy?
   end
 
   def match_account?
-    record.account_id == user.account_id
+    record.account_number == user.account_number
```

Comment:
To respond to your premature optimization comment, @vkrizan, this app-wide pundit policy calls `account_number` on some unknown model type. It's duck typing! That means that any model that pundit authorizes must respond to `account_number`. Sometimes `record` is a Host, so we can't just use `account` because of the unfortunate column name in the view. Furthermore, `account_id` is a 'magic' rails attribute that is now not available on a host (it's now `account_object_id`). We could, of course, define that method on the host to keep this working, but both solutions seem similar in terms of complexity/impact to me. I'm making the change this way looking forward to when we remove the accounts table altogether.

Reply:
For top level models that are exposed via API the account number delegate make sense. For their children, not so much. See https://github.com/RedHatInsights/compliance-backend/pull/723#discussion_r565915627 

I'd be happy to discuss pros and cons of some of the approach at some time.